packages/babel-check-binary-expressions/src/index.js

```diff
@@ -0,0 +1,8 @@
+export default function(left, right) {
+  // eslint-disable-next-line no-undef
+  if (left instanceof BigInt && right instanceof BigInt) {
+    return left.plus(right);
+  } else {
+    return left + right;
```

Comment:
I think it's not spec compliant. You need to emit TypeError if Type(left) is different of Type(right). Check the semantics here https://tc39.github.io/proposal-bigint/#sec-additive-operators

In your current function, if ```left instanceof BigInt === true``` and ```left instanceof BigInt === false``` it will fallback to else path and probably will result wrong result.